File: app/src/main/java/de/markusdamm/pokerapp/PlayerAdderToEvening.java
package de.markusdamm.pokerapp;

import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.support.v7.app.ActionBarActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;

import de.markusdamm.pokerapp.data.Player;
import de.markusdamm.pokerapp.utils.ArrayAdapterPlayer;
import de.markusdamm.pokerapp.utils.Utils;


public class PlayerAdderToEvening extends ActionBarActivity {

    private SQLiteDatabase database;
    private ListView playerLV;
    private EditText et;
    private ArrayList<Player> playerList = new ArrayList<>();
    private Button btn;
    private int eveningID;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_player_adder_to_evening);
        et = (EditText) findViewById(R.id.newPlayerName);
        Intent intent = getIntent();
        fillList();
        eveningID = intent.getIntExtra("eveId", -1);
        markUnselectablePlayers();
        addListenerOnListViewItemSelection();
    }


    public ArrayList<Player> getPlayers() {
        database = openOrCreateDatabase("pokerDB", MODE_PRIVATE, null);
        ArrayList<Player> ret = new ArrayList<>();

        Cursor cursor = database.rawQuery("SELECT id, name, gender FROM players", null);
        while (cursor.moveToNext()) {
            String entry = cursor.getString(cursor.getColumnIndex("name"));
            int id = cursor.getInt(cursor.getColumnIndex("id"));
            Player player = Utils.getPlayerFromListById(playerList, id);
            int gender = cursor.getInt(2);
            if (player == null) {
                player = new Player(entry,id,gender);
            }
            ret.add(player);
        }
        cursor.close();
        database.close();
        return ret;
    }

    public void savePlayer(View view) {
        String s = et.getText().toString();
        if (!s.equals("")) {
            Player player = new Player(s,-1,0);
            addPlayerToDB(player);
        }
        else{
            Toast.makeText(this,"Bitte Namen eingeben",Toast.LENGTH_LONG).show();
        }
    }


    public void fillList() {
        playerList = getPlayers();
        markUnselectablePlayers();
        playerLV = (ListView) findViewById(R.id.players);
        ListAdapter listenAdapter = new ArrayAdapterPlayer(this, playerList);
        playerLV.setAdapter(listenAdapter);
    }


    public void addPlayerToDB(Player player) {

        Player pl = Utils.getPlayerFromList(playerList, player.getName());
        if (pl == null) {
            database = openOrCreateDatabase("pokerDB", MODE_PRIVATE, null);
            database.execSQL("INSERT INTO players (name, gender) VALUES('" + player.getName() + "', " + player.getGenderAsInt() + ");");
            database.close();
            fillList();
            Toast.makeText(this, player.getName() + " wurde gespeichert", Toast.LENGTH_LONG).show();
        } else {
            Toast.makeText(this, "Spieler existiert schon", Toast.LENGTH_LONG).show();
        }

    }

    public void addListenerOnListViewItemSelection() {
        playerLV.setClickable(true);
        playerLV.setOnItemClickListener(new AdapterView.OnItemClickListener() {

            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Player item = (Player) playerLV.getItemAtPosition(position);
                item.toggleSelected();
                fillList();
                Toast.makeText(parent.getContext(), Integer.toString(position), Toast.LENGTH_SHORT).show();
            }
        });

    }


    public void addPlayersToEvening(View view) {
        int i = 0;
        for (Player pl : playerList) {
            if (pl.isSelected()) {
                addPlayerToEvening(pl);
                i++;
            }
        }
        Toast.makeText(this,i + " Spieler zum Abend hinzugefügt",Toast.LENGTH_LONG).show();
        finish();
    }

    public void addPlayerToEvening(Player player) {
        String sqlState = "INSERT INTO places (evening, loser, nr, winner) " +
                "VALUES(" + eveningID + ", " + Integer.toString(player.getId()) + ", -1, -1);";

        database = openOrCreateDatabase("pokerDB", MODE_PRIVATE, null);
        database.execSQL(sqlState);
        //Toast.makeText(this, sqlState, Toast.LENGTH_LONG).show();
        Cursor cursor = database.rawQuery("SELECT name FROM evenings", null);
        cursor.moveToLast();
        String entry = cursor.getString(cursor.getColumnIndex("name"));
        this.setTitle(entry);
        cursor.close();
        database.close();
        //Toast.makeText(this, entry, Toast.LENGTH_LONG).show();

    }

    public void markUnselectablePlayers() {
        database = openOrCreateDatabase("pokerDB", MODE_PRIVATE, null);
        String sqlstate = "SELECT loser FROM places WHERE evening = " + eveningID + ";";
        Cursor cursor = database.rawQuery(sqlstate, null);
        while (cursor.moveToNext()) {
            int id = cursor.getInt(0);
            Player pl = Utils.getPlayerFromListById(playerList, id);
            pl.setSelectable(false);
        }
        cursor.close();
        database.close();

    }
}

File: app/src/main/java/de/markusdamm/pokerapp/PlayerAdderToEvening.kt
package de.markusdamm.pokerapp

import android.database.sqlite.SQLiteDatabase
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import android.widget.Toast
import de.markusdamm.pokerapp.data.Player
import de.markusdamm.pokerapp.database.DatabaseHelper
import de.markusdamm.pokerapp.utils.ArrayAdapterPlayer
import de.markusdamm.pokerapp.utils.Utils
import java.util.*


class PlayerAdderToEvening : AppCompatActivity() {

    private var database: SQLiteDatabase = DatabaseHelper.getDatabase()
    private lateinit var playerLV: ListView
    private lateinit var et: EditText
    private var playerList = ArrayList<Player>()
    private val btn: Button? = null
    private var eveningID: Int = 0


    val players: ArrayList<Player>
        get() {
            val ret = ArrayList<Player>()

            val cursor = database.rawQuery("SELECT id, name, gender FROM players", null)
            while (cursor.moveToNext()) {
                val entry = cursor.getString(cursor.getColumnIndex("name"))
                val id = cursor.getInt(cursor.getColumnIndex("id"))
                var player = Utils.getPlayerById(playerList, id)
                val gender = cursor.getInt(2)
                if (player == null) {
                    player = Player(entry, id, gender)
                }
                ret.add(player)
            }
            cursor.close()
            return ret
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_player_adder_to_evening)
        et = findViewById<EditText>(R.id.newPlayerName)
        val intent = intent
        fillList()
        eveningID = intent.getIntExtra("eveId", -1)
        markUnselectablePlayers()
        addListenerOnListViewItemSelection()
    }

    fun savePlayer(view: View) {
        val s = et.text.toString()
        if (s != "") {
            val player = Player(s, -1, 0)
            addPlayerToDB(player)
        } else {
            Toast.makeText(this, "Bitte Namen eingeben", Toast.LENGTH_LONG).show()
        }
    }

    private fun fillList() {
        playerList = players
        markUnselectablePlayers()
        playerLV = findViewById<ListView>(R.id.players)
        val listenAdapter = ArrayAdapterPlayer(this, playerList)
        playerLV.adapter = listenAdapter
    }

    private fun addPlayerToDB(player: Player) {
        val pl = Utils.getPlayerByName(playerList, player.name)
        if (pl == null) {
            database.execSQL("INSERT INTO players (name, gender) VALUES('" + player.name + "', " + player.genderAsInt + ");")
            fillList()
            Toast.makeText(this, player.name + " wurde gespeichert", Toast.LENGTH_LONG).show()
        } else {
            Toast.makeText(this, "Spieler existiert schon", Toast.LENGTH_LONG).show()
        }
    }

    private fun addListenerOnListViewItemSelection() {
        playerLV.isClickable = true
        playerLV.setOnItemClickListener  { parent, _, position, _ ->
            val item = playerLV.getItemAtPosition(position) as Player
            item.toggleSelected()
            fillList()
            Toast.makeText(parent.context, position.toString(), Toast.LENGTH_SHORT).show()
        }
    }

    fun addPlayersToEvening(view: View) {
        var i = 0
        playerList.asSequence()
                .filter { it.isSelected }
                .forEach { addPlayerToEvening(it); i++ }
        Toast.makeText(this, "$i Spieler zum Abend hinzugefügt", Toast.LENGTH_LONG).show()
        finish()
    }

    private fun addPlayerToEvening(player: Player) {
        val sqlState = "INSERT INTO places (evening, loser, nr, winner) " +
                "VALUES($eveningID,${player.id}, -1, -1);"
        database.execSQL(sqlState)
        val cursor = database.rawQuery("SELECT name FROM evenings", null)
        cursor.moveToLast()
        val entry = cursor.getString(cursor.getColumnIndex("name"))
        this.title = entry
        cursor.close()
    }

    private fun markUnselectablePlayers() {
        val sqlstate = "SELECT loser FROM places WHERE evening = $eveningID;"
        val cursor = database.rawQuery(sqlstate, null)
        while (cursor.moveToNext()) {
            val id = cursor.getInt(0)
            Utils.getPlayerById(playerList, id)?.let {
                it.isSelectable = false
            }
        }
        cursor.close()
    }
}